ambry-network/src/main/java/com.github.ambry.network/NetworkMetrics.java

```diff
@@ -121,13 +125,25 @@ public NetworkMetrics(MetricRegistry registry) {
     selectorNodeMetricMap = new HashMap<String, SelectorNodeMetric>();
   }
 
-  public void initializeSelectorMetricsIfRequired(final AtomicLong activeConnections) {
+  /**
+   * Initializes a few network metrics for the selector
+   * @param activeConnections count of current active connections
+   * @param pendingHandshakeTransmissions List of {@link Transmission}s that are awaiting handshake completion
+   */
+  public void initializeSelectorMetricsIfRequired(final AtomicLong activeConnections,
+      final Map<String, PendingHandshakeTransmission> pendingHandshakeTransmissions) {
```

Comment:
technically, this can simply take `Map<String, ? extends Object>` or something and you can keep the other class private. However, that is all awkward. Would prefer if you just gave this method an `AtomicLong` like `activeConnections` and updated the `AtomicLong` where you add/remove to `transmissionsPendingHandshake`